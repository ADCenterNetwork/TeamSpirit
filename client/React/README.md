# React Frontend

/!\ From https://github.com/jgimenoa/TeamSpiritRFE

### **Discover React**:

For beginners: Learning React in 2020: https://www.youtube.com/watch?v=NJWI1b1upps&list=PLDIXF8nb0VG1v4S-smVy7GV0MHsJ3PJiL

React Tutorial - Learn React in 2020 - Building Todoist From Scratch: https://www.youtube.com/watch?v=HgfA4W_VjmI

### **Develop with React**:

There are several ways to start a React project [here](https://reactjs.org/docs/create-a-new-react-app.html).

In this project, [Create React App](https://create-react-app.dev/docs/getting-started) is used. It allows us create a project with a TypeScript setup.

### **Build and run** the app:

Follow:

- clone or download code, navigate to project's root folder
- install modules with `npm install`
- Run the App with `npm start`
- A browser window will be opened.
- Enjoy your app
