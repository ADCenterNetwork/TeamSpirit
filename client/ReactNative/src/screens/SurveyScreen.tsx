import React, { useState, useEffect } from "react";
import {
  ImageBackground,
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  BackHandler,
  Alert,
  ActivityIndicator,
} from "react-native";

import {
  heightPercentageToDP as hp,
  widthPercentageToDP as wp,
} from "react-native-responsive-screen";
import colors from "../config/colors";
import SwiperComponent from "../components/Swiper";
import SwiperCircle from "../components/SwiperCircle";
import { IQuestionStatus, IQuestionResponse } from "../models/interfaces";
import { SurveyService } from "../services/Services";
import QuestionsContext from "../context/questionsContext";

function useForceUpdate() {
  const [value, setValue] = useState(0); // integer state
  return () => setValue((value) => ++value); // update the state to force render
}

const SurveyScreen = ({
  navigation,
  route,
}: {
  navigation: any;
  route: any;
}) => {
  const forceUpdate = useForceUpdate();
  const { surveyCode, projectName, userId } = route.params;
  const surveyService: SurveyService = new SurveyService();
  const [loading, setLoading] = useState(false);

  // Function that sends the notes to the database.
  async function sendSurvey(surveyCode: string, body: IQuestionResponse[]) {
    await surveyService
      .sendSurvey(surveyCode, body)
      .then((res) => {
        if ((res.status = 200)) {
          navigation.navigate("FinalScreen", {
            message: "Thank you for submitting your answers!",
          });
        }
      })
      .catch((err) => {
        setLoading(false);
        if (err.request.status == 0) {
          Alert.alert(
            "Network Error!",
            "Please verify you have internet access.",
            [{ text: "Ok", style: "cancel" }],
            { cancelable: false }
          );
        } else {
          Alert.alert(
            "Oops!",
            err.response.data.message,
            [{ text: "Ok", style: "cancel" }],
            { cancelable: false }
          );
        }
      });
  }

  // Function that asks for your confirmation to leave when pressing the back buttton on the mobile device.
  useEffect(() => {
    const backAction = () => {
      Alert.alert(
        "Exit App",
        "Do you want to exit?",
        [
          {
            text: "No",
            style: "cancel",
          },
          { text: "Yes", onPress: () => BackHandler.exitApp() },
        ],
        { cancelable: false }
      );
      return true;
    };
    const backHandler = BackHandler.addEventListener(
      "hardwareBackPress",
      backAction
    );
    return () => backHandler.remove();
  }, []);
  const handleSurveyCompletion = () => {
    setLoading(true);
    sendSurvey(surveyCode.code, questionsResponse);
  };
  const [currentQuestion, setCurrentQuestion] = useState(0);
  const [disabled, setDisabled] = useState(true);
  const [questionsState, setQuestionsState] = useState<IQuestionStatus[]>([
    { valid: false, status: "active", touched: true },
    { valid: false, status: "", touched: false },
    { valid: true, status: "", touched: false },
    { valid: true, status: "", touched: false },
    { valid: false, status: "", touched: false },
    { valid: false, status: "", touched: false },
  ]);

  var Hashes = require("jshashes");
  var SHA256 = new Hashes.SHA256();

  const uniqueUserId: any = userId;

  const [questionsResponse, setQuestionsResponse] = useState<
    IQuestionResponse[]
  >([
    {
      number: 1,
      note: 0,
      surveyCode: surveyCode.code,
      user: SHA256.hex(uniqueUserId),
    },
    {
      number: 2,
      note: 0,
      surveyCode: surveyCode.code,
      user: SHA256.hex(uniqueUserId),
    },
    {
      number: 3,
      note: 5,
      surveyCode: surveyCode.code,
      user: SHA256.hex(uniqueUserId),
    },
    {
      number: 4,
      note: 5,
      surveyCode: surveyCode.code,
      user: SHA256.hex(uniqueUserId),
    },
    {
      number: 5,
      note: 0,
      surveyCode: surveyCode.code,
      user: SHA256.hex(uniqueUserId),
    },
    {
      number: 6,
      note: 0,
      surveyCode: surveyCode.code,
      user: SHA256.hex(uniqueUserId),
    },
  ]);

  const [activeIcon, setActiveIcon] = useState([
    [false, false, false, false, false],
    [false, false, false, false, false],
    [],
    [],
    [],
    [false, false],
  ]);

  // Updates the cycle of life of the application.
  useEffect(() => {
    let count = 0;
    const currentQuestionStatus = questionsState;
    currentQuestionStatus.map((question: IQuestionStatus) => {
      question.status = "";
      if (question.valid) {
        count++;
      }
    });
    currentQuestionStatus[currentQuestion].status = "active";
    currentQuestionStatus[currentQuestion].touched = true;
    setQuestionsState(currentQuestionStatus);
    const response = questionsResponse;
    setQuestionsResponse(response);
    if (count === questionsState.length) {
      setDisabled(false);
      forceUpdate();
    }
    forceUpdate();
  }, [questionsState, activeIcon, currentQuestion]);

  return (
    <QuestionsContext.Provider
      value={{
        questionsResponse: questionsResponse,
        setQuestionsResponse: setQuestionsResponse,
        disabled: disabled,
        setDisabled: setDisabled,
      }}
    >
      <ImageBackground
        source={require("../assets/surveyBackfround.png")}
        style={styles.container}
      >
        {loading ? (
          <View style={styles.activityIndicatorBackground}>
            <ActivityIndicator
              size="large"
              color={colors.primary}
              style={styles.activityIndicator}
            />
          </View>
        ) : null}
        <View style={styles.projectView}>
          <Text style={styles.project}>{projectName}</Text>
        </View>
        <View style={styles.swiper}>
          <SwiperComponent
            activeIcon={activeIcon}
            setActiveIcon={setActiveIcon}
            setCurrentQuestion={setCurrentQuestion}
            setQuestionsState={setQuestionsState}
            questionsState={questionsState}
            useForceUpdate={useForceUpdate}
          />
          <View style={styles.circle}>
            <SwiperCircle state={questionsState} />
          </View>
        </View>
        <TouchableOpacity
          style={disabled ? styles.btnDisabled : styles.btn}
          activeOpacity={0.5}
          onPress={handleSurveyCompletion}
          disabled={disabled}
        >
          <Text style={styles.btnTitle}>Submit</Text>
        </TouchableOpacity>
      </ImageBackground>
    </QuestionsContext.Provider>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.white,
    alignItems: "center",
    justifyContent: "center",
  },
  project: {
    color: colors.primary,
    fontSize: 20,
    fontWeight: "bold",
    textAlign: "left",
  },
  projectView: {
    position: "relative",
    top: -64,
    marginLeft: wp("-50%"),
    width: wp("40%"),
    paddingLeft: 0,
  },
  swiper: {
    flex: 0.7,
    bottom: "14%",
  },
  circle: {
    alignItems: "center",
    justifyContent: "center",
    top: hp("0%"),
    left: wp("30%"),
    width: wp("40%"),
    backgroundColor: colors.transparent,
  },
  btn: {
    backgroundColor: colors.primary,
    width: wp("40%"),
    alignItems: "center",
    justifyContent: "center",
    height: hp("7%"),
    borderRadius: 30,
  },
  btnDisabled: {
    backgroundColor: colors.disabled,
    width: wp("40%"),
    alignItems: "center",
    justifyContent: "center",
    height: hp("7%"),
    borderRadius: 30,
  },
  btnTitle: {
    color: colors.white,
    fontSize: 20,
  },
  activityIndicator: {
    margin: "auto",
  },
  activityIndicatorBackground: {
    zIndex: 150,
    backgroundColor: "rgba(0, 0, 0, 0.3)",
    position: "absolute",
    top: 0,
    left: 0,
    width: wp("100%"),
    height: hp("100%"),
    justifyContent: "center",
  },
});

export default SurveyScreen;
