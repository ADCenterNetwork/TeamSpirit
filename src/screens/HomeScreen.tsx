import React, { useState } from "react";
import {
  StyleSheet,
  TextInput,
  Text,
  View,
  Alert,
  ActivityIndicator,
  TouchableOpacity,
  Keyboard,
  ImageBackground,
  KeyboardAvoidingView,
  TouchableWithoutFeedback,
} from "react-native";

import colors from "../config/colors";
import { CodeValidationService } from "../services/Services";
import { IValidationCode } from "../models/interfaces";

const HomeScreen = ({ navigation }: { navigation: any }) => {
  const [inputText, setInputText] = useState("");
  const [loading, setLoading] = useState(false);
  const codeValidationService: CodeValidationService = new CodeValidationService();

  const resetInputHandler = () => {
    setLoading(false);
    setInputText("");
  };

  async function sendCode(inputText: IValidationCode) {
    await codeValidationService
      .sendCode(inputText)
      .then((res) => {
        if (res.status == 200) {
          navigation.navigate("SurveyScreen", {
            surveyCode: inputText,
            projectName: res.data.TeamName,
          });
        }
      })
      .catch((err) => {
        resetInputHandler();
        if (err.request.status == 0) {
          Alert.alert(
            "Network Error!",
            "Please verify you have internet access.",
            [{ text: "Ok", style: "cancel" }],
            { cancelable: false }
          );
        } else {
          Alert.alert(
            "Oops!",
            err.response.data.message,
            [{ text: "Ok", style: "cancel" }],
            { cancelable: false }
          );
        }
      });
  }

  const submitHandler = () => {
    setLoading(true);
    Keyboard.dismiss();
    sendCode({ code: inputText });
  };

  return (
    <KeyboardAvoidingView style={styles.container}>
      <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
        <View style={styles.container}>
          <ImageBackground
            source={require("../assets/homeBackground.png")}
            style={styles.imgBackground}
          >
            {loading ? (
              <View style={styles.activityIndicatorBackground}>
                <ActivityIndicator
                  size="large"
                  color={colors.primary}
                  style={styles.activityIndicator}
                />
              </View>
            ) : null}
            <View style={styles.contentAbove}>
              <Text style={styles.title}>Welcome to Team Spirit Survey!</Text>
            </View>
            <View style={styles.contentBelow}>
              <Text style={styles.textSentence1}>
                Please, answer the following 6 questions. It will take you a
                couple of minutes.
              </Text>
              <Text style={styles.textSentence2}>
                But first, write the code of your team:
              </Text>
              <TextInput
                style={styles.input}
                placeholder="My project is..."
                onChangeText={(inputText) => setInputText(inputText)}
                value={inputText}
                maxLength={10}
              />

              <TouchableOpacity onPress={submitHandler} style={styles.submit}>
                <Text style={styles.insideButton}>Start</Text>
              </TouchableOpacity>
            </View>
          </ImageBackground>
        </View>
      </TouchableWithoutFeedback>
    </KeyboardAvoidingView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.white,
  },
  contentAbove: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  contentBelow: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    marginBottom: 50,
  },
  title: {
    position: "absolute",
    width: "90%",
    top: "50%",
    fontFamily: "Roboto",
    fontStyle: "normal",
    fontWeight: "bold",
    fontSize: 22,
    lineHeight: 23,
    textAlign: "center",
    color: colors.primary,
  },
  textSentence1: {
    width: "90%",
    fontFamily: "Roboto",
    fontStyle: "normal",
    fontWeight: "normal",
    fontSize: 14,
    lineHeight: 16,
    textAlign: "center",
    color: colors.white,
    marginBottom: 30,
  },
  textSentence2: {
    width: "90%",
    fontFamily: "Roboto",
    fontStyle: "normal",
    fontWeight: "normal",
    fontSize: 14,
    lineHeight: 16,
    textAlign: "center",
    color: colors.white,
  },
  input: {
    width: "45%",
    height: 30,
    backgroundColor: colors.white,
    borderRadius: 20,
    marginBottom: 30,
    marginTop: 30,
    paddingLeft: 10,
  },
  submit: {
    width: "45%",
    height: 70,
    borderColor: colors.white,
    borderWidth: 1,
    borderRadius: 50,
  },
  imgBackground: {
    flex: 1,
    resizeMode: "contain",
    justifyContent: "center",
  },
  insideButton: {
    textAlign: "center",
    fontSize: 24,
    lineHeight: 65,
    color: colors.white,
  },
  activityIndicator: {
    margin: "auto",
  },
  activityIndicatorBackground: {
    zIndex: 150,
    backgroundColor: "rgba(0, 0, 0, 0.3)",
    position: "absolute",
    top: 0,
    left: 0,
    width: "100%",
    height: "100%",
    justifyContent: "center",
  },
});

export default HomeScreen;
