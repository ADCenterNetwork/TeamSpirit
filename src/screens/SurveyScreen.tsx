import React, { useState, useEffect } from "react";
import {
  ImageBackground,
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
} from "react-native";

import colors from "../config/colors";
import SwiperComponent from "../components/Swiper";
import SwiperCircle from "../components/SwiperCircle";
import { IQuestionStatus } from "../models/interfaces";

const SurveyScreen = ({ navigation }: { navigation: any }) => {
  const handleSurveyCompletion = () => navigation.navigate("SuccessScreen");
  const [currentQuestion, setCurrentQuestion] = useState(0);
  const [disabled, setDisabled] = useState(true);
  const [questionsState, setQuestionsState] = useState<IQuestionStatus[]>([
    { valid: false, answer: 0, status: "active", touched: false },
    { valid: false, answer: 0, status: "", touched: false },
    { valid: true, answer: 5, status: "", touched: false },
    { valid: true, answer: 5, status: "", touched: false },
    { valid: false, answer: 0, status: "", touched: false },
    { valid: false, answer: 0, status: "", touched: false },
  ]);
  const [activeIcon, setActiveIcon] = useState([
    [false, false, false, false, false],
    [false, false, false, false, false],
    [],
    [],
    [],
    [false, false],
  ]);

  useEffect(() => {
    let count = 0;
    const currentQuestionStatus = questionsState;
    currentQuestionStatus.map((question: IQuestionStatus) => {
      question.status = "";
      if (question.valid) {
        count++;
      }
    });
    currentQuestionStatus[currentQuestion].status = "active";
    currentQuestionStatus[currentQuestion].touched = true;
    setQuestionsState(currentQuestionStatus);
    if (count === questionsState.length) {
      setDisabled(false);
    }

    console.log(questionsState);
  });

  return (
    <ImageBackground
      source={require("../assets/surveyBackfround.png")}
      style={styles.container}
    >
      <Text style={styles.project}>Project's Name</Text>
      <View style={styles.swiper}>
        <SwiperComponent
          activeIcon={activeIcon}
          setActiveIcon={setActiveIcon}
          setCurrentQuestion={setCurrentQuestion}
          setQuestionsState={setQuestionsState}
          questionsState={questionsState}
        />
        <View style={styles.circle}>
          <SwiperCircle state={questionsState} />
        </View>
      </View>
      <TouchableOpacity
        style={disabled ? styles.btnDisabled : styles.btn}
        activeOpacity={0.5}
        onPress={handleSurveyCompletion}
        disabled={disabled}
      >
        <Text style={styles.btnTitle}>Submit</Text>
      </TouchableOpacity>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.white,
    alignItems: "center",
    justifyContent: "center",
  },
  project: {
    flex: 0.1,
    right: "27%",
    bottom: "5%",
    color: colors.primary,
    fontSize: 20,
    fontWeight: "bold",
  },
  swiper: {
    flex: 0.7,
    bottom: "16%",
  },
  circle: {
    alignItems: "center",
    justifyContent: "center",
  },
  btn: {
    backgroundColor: colors.primary,
    width: 160,
    alignItems: "center",
    justifyContent: "center",
    height: 60,
    borderRadius: 30,
  },
  btnDisabled: {
    backgroundColor: colors.disabled,
    width: 160,
    alignItems: "center",
    justifyContent: "center",
    height: 60,
    borderRadius: 30,
  },
  btnTitle: {
    color: colors.white,
    fontSize: 20,
  },
});

export default SurveyScreen;
