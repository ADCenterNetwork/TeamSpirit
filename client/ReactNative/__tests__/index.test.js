import React from "react";
import render from "react-test-renderer";
import HomeScreen from "../src/screens/HomeScreen";
import SurveyScreen from "../src/screens/SurveyScreen";

// render() renders the component to an in-memory representation that doesn't require an iOS or Android environment.
//queryByText() finds a child component that contains the passed-in text.
//

test("HomeScreen snapShot", () => {
  jest.useFakeTimers();
  const snap = render.create(<HomeScreen />).toJSON();
  expect(snap).toMatchSnapshot();
});

/* test("SurveyScreen snapShot", () => {
  const snap = render.create(<SurveyScreen />).toJSON();
  expect(snap).toMatchSnapshot(); 
});
*/
