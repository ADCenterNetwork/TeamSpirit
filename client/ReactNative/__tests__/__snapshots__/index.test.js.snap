// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapShot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "resizeMode": "contain",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "resizeMode": "contain",
      }
    }
  >
    <Image
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#79C0C6",
            "fontFamily": "Roboto",
            "fontSize": 22,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": 23,
            "position": "absolute",
            "textAlign": "center",
            "top": "50%",
            "width": "90%",
          }
        }
      >
        Welcome to Team Spirit Survey!
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginBottom": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "Roboto",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": 16,
            "marginBottom": 30,
            "textAlign": "center",
            "width": "90%",
          }
        }
      >
        Please, answer the following 6 questions. It will take you a couple of minutes.
      </Text>
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "Roboto",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": 16,
            "textAlign": "center",
            "width": "90%",
          }
        }
      >
        But first, write the code of your team:
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="characters"
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="My project is..."
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 20,
            "height": 30,
            "marginBottom": 30,
            "marginTop": 30,
            "paddingLeft": 10,
            "width": "45%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#fff",
            "borderRadius": 50,
            "borderWidth": 1,
            "height": 120,
            "opacity": 1,
            "width": 337.5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 40,
              "paddingBottom": 21.5,
              "paddingTop": 29.5,
              "textAlign": "center",
            }
          }
        >
          Start
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`HomeScreen snapShot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "resizeMode": "contain",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "resizeMode": "contain",
      }
    }
  >
    <Image
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#79C0C6",
            "fontFamily": "Roboto",
            "fontSize": 22,
            "fontStyle": "normal",
            "fontWeight": "bold",
            "lineHeight": 23,
            "position": "absolute",
            "textAlign": "center",
            "top": "50%",
            "width": "90%",
          }
        }
      >
        Welcome to Team Spirit Survey!
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginBottom": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "Roboto",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": 16,
            "marginBottom": 30,
            "textAlign": "center",
            "width": "90%",
          }
        }
      >
        Please, answer the following 6 questions. It will take you a couple of minutes.
      </Text>
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "Roboto",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "normal",
            "lineHeight": 16,
            "textAlign": "center",
            "width": "90%",
          }
        }
      >
        But first, write the code of your team:
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="characters"
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="My project is..."
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 20,
            "height": 30,
            "marginBottom": 30,
            "marginTop": 30,
            "paddingLeft": 10,
            "width": "45%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#fff",
            "borderRadius": 50,
            "borderWidth": 1,
            "height": 120,
            "opacity": 1,
            "width": 337.5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 40,
              "paddingBottom": 21.5,
              "paddingTop": 29.5,
              "textAlign": "center",
            }
          }
        >
          Start
        </Text>
      </View>
    </View>
  </View>
</View>
`;
